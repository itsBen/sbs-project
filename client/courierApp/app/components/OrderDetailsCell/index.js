import React from 'react';
import { View, Text } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { capitalize } from '@utilities';

export default ({ label, info, iconName }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.labelText}>
        <Icon name={iconName} color="#353535" size={14} /> {label}
      </Text>
      <Text style={styles.infoText}>{info && capitalize(info)}</Text>
    </View>
  );
};

const styles = {
  container: {
    flexDirection: 'column',
    paddingVertical: 8,
  },
  labelText: {
    fontSize: 15,
    lineHeight: 25,
    color: '#353535',
  },
  infoText: {
    fontSize: 17,
  },
};
