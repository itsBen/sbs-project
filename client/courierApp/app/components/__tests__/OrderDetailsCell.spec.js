/* eslint-env node, jest */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import OrderDetailsCell from '../OrderDetailsCell';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <OrderDetailsCell
        label="test label"
        info="some info"
        iconName="location-on"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
