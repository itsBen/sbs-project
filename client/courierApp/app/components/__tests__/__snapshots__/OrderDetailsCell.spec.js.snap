// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "paddingVertical": 8,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#353535",
        "fontSize": 15,
        "lineHeight": 25,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#353535",
            "fontSize": 14,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      îƒˆ
    </Text>
     
    test label
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 17,
      }
    }
  >
    Some info
  </Text>
</View>
`;
